### DISCUSSION TITLE
Fire ar-status event for native apps

### DISCUSSION THREAD

**Discussion by useronym** [google/model-viewer#2235] [Q&A]
### Description

Window on focus event is not fired when the model viewer AR window is closed and you're returned back to the webpage you launched it from. Or is there another way to detect that the model viewer webxr window has been closed? On iOS, as soon as the QuickLook AR preview is closed, this event is fired.

#### Live Demo
https://glitch.com/edit/#!/pointy-fantasy-moon

### Version
- model-viewer: v1.2.1

### Browser Affected
- [x] Chrome (v1.2.1)
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS
- [x] Android 11
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by useronym** [comment]
Okay, looks like this can be resolved by adding a listener for the `ar-status` event and check that the `event.details.status` is `not-presenting`. Downside being that this doesn't work on iOS, of course. Feel free to close this issue, but it might be worthwhile to try to present some kind of unified interface across platforms for detecting that the AR session has ended?

**Comment by elalish** [comment]
Indeed, you've found our solution. That is interesting that you're using the focus event for detecting the native app returning. If that's reliable enough across iOS and Android I'd consider firing `ar-status` based on that.

**Comment by useronym** [comment]
It sort-of makes sense on iOS that it would fire the window focus event, because I think Safari launches a separate app to handle the quicklook preview functionality. But, of course, with webXR we stay on the webpage, so no onfocus event gets fired.
